chapters/connectors.tex

```diff
@@ -903,24 +905,58 @@ \subsection{Connection Graphs and Their Operators}\label{overconstrained-equatio
+\end{nonnormative}
+The selected optional spanning tree edges, together with all required spanning tree edges generated from \lstinline!Connections.branch!, and nodes corresponding to definite and potential roots from the virtual connection graph.
+
+\subsubsection{Spanning Trees}
 Before connection equations are generated, the virtual connection graph is transformed into a set of spanning trees by removing optional spanning tree edges from the graph.
 This is performed in the following way:
 \begin{enumerate}
 \item
-  Every root node defined via the \lstinline!Connections.root!-equation is a definite root of one spanning tree.
+  Root nodes are selected as follows:
+\begin{enumerate}
+\item
+  Every definite root node defined via the \lstinline!Connections.root!-equation is a root of one spanning tree.
+  It is an error if two (or more) definite root nodes are connected through required spanning tree edges.
 \item
   The virtual connection graph may consist of sets of subgraphs that are not connected together.
-  Every subgraph in this set shall have at least one root node or one potential root node in a simulation model.
-  If a graph of this set does not contain any root node, then one potential root node in this subgraph that has the lowest priority number is selected to be the root of that subgraph.
+  Every subgraph in this set shall have at least one definite root node or one potential root node in a simulation model.
+  If a graph of this set does not contain any definite root node, then one potential root node in this subgraph that has the lowest priority number is selected to be the root of that subgraph.
   The selection can be inquired in a class with function \lstinline!Connections.isRoot!, see table above.
+\end{enumerate}
+\item
+  If there is a cycle among required spanning-tree-edges it is an error, as it is not possible to construct a spanning tree.
 \item
-  If there are n selected roots in a subgraph, then optional spanning-tree edges have to be removed such that the result shall be a set of n spanning trees with the selected root nodes as roots.
+  For a subgraph with $n$ selected roots, optional spanning-tree edges are removed such that the result is a set of $n$ spanning trees with the selected root nodes as roots.
 \end{enumerate}
 
+\subsubsection{Equations}
 After this analysis, the \firstuse{connection graph equations}\index{connection equations!connection graph} are generated in the following way:
 \begin{enumerate}
 \item
-  For every optional spanning-tree edge (i.e., a \lstinline!connect(A, B)! equation), in one of the spanning trees, the connection equations are generated according to \cref{generation-of-connection-equations}.
+  For every optional spanning-tree edge in any of the spanning trees, the connection equations are generated according to \cref{generation-of-connection-equations}.
```

Comment:
Extra clarity:
```suggestion
  For every remaining optional spanning-tree edge in any of the spanning trees, the connection equations are generated according to \cref{generation-of-connection-equations}.
```

Reply:
On this line I don't think that remaining is that helpful.
However, it makes it consistent with the ones that aren't in the spanning trees below.